static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -2,179 +2,233 @@ define([
     'common/utils/fastdom-promise',
     'Promise',
     'common/utils/$',
+    'common/utils/create-store',
     'common/utils/mediator',
     'common/utils/config',
-    'common/utils/detect'
+    'common/utils/detect',
+    'lodash/objects/assign'
 ], function (
     fastdom,
     Promise,
     $,
+    createStore,
     mediator,
     config,
-    detect) {
+    detect,
+    assign) {
+
+    var $adBanner = $('.top-banner-ad-container--above-nav');
+    var $adBannerInner = $('.ad-slot--top-above-nav', $adBanner);
+    var $header = $('.js-header');
+    var $document = $(document.body);
+
+    var getClientAdHeight = function () {
+        return fastdom.read(function () {
+            return $adBannerInner[0].clientHeight;
+        });
+    };
 
-    return function () {
-        if (detect.getBreakpoint() !== 'mobile' && config.page.contentType !== 'Interactive') {
-            fastdom.read(function () {
-                var adId = 'dfp-ad--top-above-nav',
-                    $adBanner = $('.js-top-banner-above-nav'),
-                    $adBannerInner = $('#' + adId, $adBanner),
-                    $header = $('.js-header');
-
-                var getClientAdHeight = function () {
-                    return fastdom.read(function () {
-                        return $adBannerInner[0].clientHeight;
+    var getAdIframe = function () { return $('iframe', $adBanner); };
+
+    var topAdRenderedPromise = new Promise(function (resolve) {
+        mediator.on('modules:commercial:dfp:rendered', function (event) {
+            var dfpAdSlotId = 'dfp-ad--top-above-nav';
+            var isEventForTopAdBanner = event.slot.getSlotElementId() === dfpAdSlotId;
+            if (isEventForTopAdBanner) { resolve(); }
+        });
+    });
+
+    var newRubiconAdHeightPromise = new Promise(function (resolve) {
+        window.addEventListener('message', function (event) {
+            var data;
+
+            // other DFP events get caught by this listener, but if they're not json we don't want to parse them or use them
+            try {
+                data = JSON.parse(event.data);
+            } catch (e) {/**/}
+
+            if (data) {
+                var $iframe = getAdIframe();
+                var isRubiconAdEvent = data.type === 'set-ad-height';
+                var isEventForTopAdBanner = isRubiconAdEvent && data.value.id === $iframe[0].id;
+
+                if (isRubiconAdEvent && isEventForTopAdBanner) {
+                    fastdom.read(function () {
+                        var padding = parseInt($adBannerInner.css('padding-top'))
+                            + parseInt($adBannerInner.css('padding-bottom'));
+                        var clientHeight = parseInt(data.value.height) + padding;
+                        resolve(clientHeight);
                     });
-                };
+                }
+            }
+        });
+    });
```

Comment:
@regiskuckaertz I wonder if this would be helpful to you. Your `trackAd` will tell you when DFP renders, but Rubicon ads (which are loaded through DFP) can render themselves again asynchronously, which is what this promise accounts for.